src/runtime_src/core/tools/xbmgmt2/flash/firmware_image.cpp

```diff
@@ -431,37 +441,34 @@ std::vector<DSAInfo> firmwareImage::getIntalledDSAs()
     }
 
     // for 2RP
-    boost::filesystem::path p = FORMATTED_FW_DIR;
-    if (!boost::filesystem::is_directory(p))
-        return installedDSA;
-
-    boost::filesystem::path formatted_fw_dir(FORMATTED_FW_DIR);
-
-    for (const std::string& t : { XSABIN_FILE_SUFFIX, DSABIN_FILE_SUFFIX }) {
-
-        std::regex e("^" FORMATTED_FW_DIR "/([^/]+)/([^/]+)/([^/]+)/.+\\." + t);
-        std::smatch cm;
-
-        for (boost::filesystem::recursive_directory_iterator iter(formatted_fw_dir,
-            boost::filesystem::symlink_option::recurse), recursive_end; iter != recursive_end;) {
-            std::string name = iter->path().string();
-            std::regex_match(name, cm, e);
-            if (cm.size() > 0) {
-                std::string pr_board = cm.str(1);
-                std::string pr_family = cm.str(2);
-                std::string pr_name = cm.str(3);
-                DSAInfo dsa(name, pr_board, pr_family, pr_name);
+    boost::filesystem::path lin_2rp(FORMATTED_FW_DIR);
+    boost::filesystem::path win_2rp(FORMATTED_FW_WIN_DIR);
+	  boost::filesystem::path formatted_fw_dir;
+	  if (boost::filesystem::is_directory(lin_2rp)) {
+		    formatted_fw_dir = lin_2rp;
```

Comment:
Is there a reason why this can cannot be written as follows:
`formatted_fw_dir = boost::filesystem::path(FORMATTED_FW_DIR);`

If so, could you remove lines 444 & 445 and update lines 450.